geode-assembly/src/test/java/org/apache/geode/rest/internal/web/RestSecurityPostProcessorTest.java

```diff
@@ -0,0 +1,185 @@
+   */
+  @Test
+  public void getRegionKey() throws Exception {
+
+    // Test a single key
+    HttpResponse response = restClient.doGet("/" + REGION_NAME + "/key1", "key1User", "1234567");
+    assertEquals(200, getCode(response));
+    assertEquals(MediaType.APPLICATION_JSON_UTF8_VALUE, getContentType(response));
+
+    String body = IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+    assertTrue(body.startsWith("\"key1User/" + REGION_NAME + "/key1/"));
+
+    // Test multiple keys
+    response = restClient.doGet("/" + REGION_NAME + "/key1,key2", "dataReader", "1234567");
+    assertEquals(200, getCode(response));
+    assertEquals(MediaType.APPLICATION_JSON_UTF8_VALUE, getContentType(response));
+
+    JSONObject jsonObject = getJsonObject(response);
+    JSONArray jsonArray = jsonObject.getJSONArray(REGION_NAME);
+    final int length = jsonArray.length();
+    for (int index = 0; index < length; ++index) {
+      String data = jsonArray.getString(index);
+      assertTrue(data.contains("dataReader/" + REGION_NAME + "/"));
+    }
+  }
+
+  @Test
+  public void getRegion() throws Exception {
+    HttpResponse response = restClient.doGet("/" + REGION_NAME, "dataReader", "1234567");
+    assertEquals(200, getCode(response));
+    assertEquals(MediaType.APPLICATION_JSON_UTF8_VALUE, getContentType(response));
+
+    JSONObject jsonObject = getJsonObject(response);
+    JSONArray jsonArray = jsonObject.getJSONArray(REGION_NAME);
+    final int length = jsonArray.length();
+    for (int index = 0; index < length; ++index) {
+      String data = jsonArray.getString(index);
+      assertTrue(data.contains("dataReader/" + REGION_NAME + "/"));
+    }
+  }
+
+  @Test
+  public void adhocQuery() throws Exception {
+    String query =
+        "/queries/adhoc?q=" + URLEncoder.encode("SELECT * FROM /" + REGION_NAME, "UTF-8");
+    HttpResponse response = restClient.doGet(query, "dataReader", "1234567");
+    assertEquals(200, getCode(response));
+    assertEquals(MediaType.APPLICATION_JSON_UTF8_VALUE, getContentType(response));
+
+    JSONArray jsonArray = getJsonArray(response);
+    final int length = jsonArray.length();
+    for (int index = 0; index < length; ++index) {
+      String data = jsonArray.getString(index);
+      assertTrue(data.contains("dataReader/" + REGION_NAME + "/"));
+    }
+  }
+
+  @Test
+  public void namedQuery() throws Exception {
+    String namedQuery = "SELECT c FROM /customers c WHERE c.customerId = $1";
+
+    HttpResponse response =
+        restClient.doPost("/queries?id=selectCustomer&q=" + URLEncoder.encode(namedQuery, "UTF-8"),
+            "dataReader", "1234567", "");
+    assertEquals(201, getCode(response));
+
+    String query = "/queries";
+    response = restClient.doGet(query, "dataReader", "1234567");
+    assertEquals(200, getCode(response));
+    assertEquals(MediaType.APPLICATION_JSON_UTF8_VALUE, getContentType(response));
+
+    String customerJson =
```

Comment:
This breaks from the pattern RestAPIsQueryAndFEJUnitTest follows.  But as you said in an earlier comment about the Orders, it will read cleaner.